lib/chef/provider/aws_vpc.rb

```diff
@@ -18,12 +19,12 @@ def action_create
 
     # Replace the main route table for the VPC
     if !new_resource.main_route_table.nil?
-      main_route_table = update_main_route_table(vpc)
+      update_main_route_table(vpc, new_resource.main_route_table)
```

Comment:
TODO - don't do this, update whatever the main route table is, don't create an implicit route table